internal/provider/data_source_local_command.go

```diff
@@ -0,0 +1,227 @@
+					"populated regardless of the exit code returned.",
+				Computed: true,
+			},
+		},
+	}
+}
+
+type localCommandDataSourceModel struct {
+	Command              types.String `tfsdk:"command"`
+	Arguments            types.List   `tfsdk:"arguments"`
+	Stdin                types.String `tfsdk:"stdin"`
+	WorkingDirectory     types.String `tfsdk:"working_directory"`
+	AllowNonZeroExitCode types.Bool   `tfsdk:"allow_non_zero_exit_code"`
+	ExitCode             types.Int64  `tfsdk:"exit_code"`
+	Stdout               types.String `tfsdk:"stdout"`
+	Stderr               types.String `tfsdk:"stderr"`
+}
+
+func (a *localCommandDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state localCommandDataSourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Prep the command
+	command := state.Command.ValueString()
+	if _, err := exec.LookPath(command); err != nil {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("command"),
+			"Command Lookup Failed",
+			"The data source received an unexpected error while attempting to find the command."+
+				"\n\n"+
+				"The command must be accessible according to the platform where Terraform is running."+
+				"\n\n"+
+				"If the expected command should be automatically found on the platform where Terraform is running, "+
+				"ensure that the command is in an expected directory. On Unix-based platforms, these directories are "+
+				"typically searched based on the '$PATH' environment variable. On Windows-based platforms, these directories "+
+				"are typically searched based on the '%PATH%' environment variable."+
+				"\n\n"+
+				"If the expected command is relative to the Terraform configuration, it is recommended that the command name includes "+
+				"the interpolated value of 'path.module' before the command name to ensure that it is compatible with varying module usage. For example: \"${path.module}/my-command\""+
+				"\n\n"+
+				"The command must also be executable according to the platform where Terraform is running. On Unix-based platforms, the file on the filesystem must have the executable bit set. "+
+				"On Windows-based platforms, no action is typically necessary."+
+				"\n\n"+
+				fmt.Sprintf("Platform: %s\n", runtime.GOOS)+
+				fmt.Sprintf("Command: %s\n", command)+
+				fmt.Sprintf("Error: %s", err),
+		)
+		return
+	}
+
+	arguments := make([]string, 0)
+	for _, element := range state.Arguments.Elements() {
+		strElement, ok := element.(types.String)
+		// Mirroring the underlying os/exec Command support for args (no nil arguments, but does support empty strings)
+		if element.IsNull() || !ok {
+			continue
+		}
+
+		arguments = append(arguments, strElement.ValueString())
+	}
+
+	cmd := exec.CommandContext(ctx, command, arguments...)
+
+	cmd.Dir = state.WorkingDirectory.ValueString()
```

Comment:
Wondering if it's worth performing some validation on the working directory e.g. is it a valid path, does it exist similar to how we're checking that the executable exists and is accessible. Or is this already adequately handled by an error returned from executing the command?

One more thought is whether it's worth mentioning that this needs to be an absolute path or whether relative paths are supported.